jaxrs/src/main/java/com/webcohesion/enunciate/modules/jaxrs/JaxrsModule.java

```diff
@@ -16,241 +14,285 @@
-          || (ApplicationPath.class.getName().equals(classAnnotation))) {
-          return true;
-        }
-      }
-    }
-    return false;
-  }
-
-  public class MediaTypeDependencySpec implements DependencySpec {
-
-    @Override
-    public boolean accept(EnunciateModule module) {
-      if (module instanceof MediaTypeDefinitionModule) {
-        MediaTypeDefinitionModule definitionModule = (MediaTypeDefinitionModule) module;
-        mediaTypeModules.add(definitionModule);
-
-        // suggest to the media type definition module that it should take a passive approach to detecting data types
-        // because this module will be aggressively adding the data type definitions to it.
-        definitionModule.setDefaultDataTypeDetectionStrategy(DataTypeDetectionStrategy.passive);
-        return true;
-      }
-
-      return false;
-    }
-
-    @Override
-    public boolean isFulfilled() {
-      // this spec is always fulfilled.
-      return true;
-    }
-
-    @Override
-    public String toString() {
-      return "media type definition modules";
-    }
-  }
+@SuppressWarnings("unchecked")
+public class JaxrsModule extends BasicEnunicateModule
+		implements TypeFilteringModule, ApiRegistryProviderModule, ApiFeatureProviderModule {
+
+	private DataTypeDetectionStrategy defaultDataTypeDetectionStrategy;
+	private final List<MediaTypeDefinitionModule> mediaTypeModules = new ArrayList<MediaTypeDefinitionModule>();
+	private ApiRegistry apiRegistry;
+	static final String NAME = "jaxrs";
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public List<DependencySpec> getDependencySpecifications() {
+		return Arrays.asList((DependencySpec) new MediaTypeDependencySpec());
+	}
+
+	public DataTypeDetectionStrategy getDataTypeDetectionStrategy() {
+		String dataTypeDetection = this.config.getString("[@datatype-detection]", null);
+
+		if (dataTypeDetection != null) {
+			try {
+				return DataTypeDetectionStrategy.valueOf(dataTypeDetection);
+			} catch (IllegalArgumentException e) {
+				// fall through...
+			}
+		}
+
+		if (this.defaultDataTypeDetectionStrategy != null) {
+			return this.defaultDataTypeDetectionStrategy;
+		}
+
+		if (this.enunciate.getIncludePatterns().isEmpty()) {
+			// if there are no configured include patterns, then we'll just
+			// stick with "local" detection so we don't include too much.
+			return DataTypeDetectionStrategy.local;
+		} else {
+			// otherwise, we'll assume the user knows what (s)he's doing and
+			// aggressively include everything.
+			return DataTypeDetectionStrategy.aggressive;
+		}
+	}
+
+	public void setDefaultDataTypeDetectionStrategy(DataTypeDetectionStrategy strategy) {
+		this.defaultDataTypeDetectionStrategy = strategy;
+	}
+
+	@Override
+	public void setApiRegistry(ApiRegistry registry) {
+		this.apiRegistry = registry;
+	}
+
+	public EnunciateJaxrsContext getJaxrsContext() {
+		return new EnunciateJaxrsContext(context);
```

Comment:
this is wrong. i dont need this method, because we dont need the client jars. Well we must return a list of context now^^